src/main.ts

```diff
@@ -35,10 +38,56 @@ app.on("ready", () =>
       // dialog.showMessageBox({ message: process.argv.join("\n"), buttons: ["OK"] })
    }
 
-   const selectedFolder = dialog.showOpenDialogSync({properties: ["openDirectory"]}) ?? []
-
-   if (selectedFolder.length === 0) {
-      app.quit()
+   // TODO: If there is not a saved folder path or it does not exist, then show the dialog to allow the user to select it...   
+   const VALID_IMAGE_EXTENSIONS = [".jpg", ".jpeg", ".png", ".gif", ".bmp", ".webp"]
+   function isFolderFullOfImages(folderPath: string): boolean {
+      const files = readdirSync(folderPath)
+         .filter(file => {
+            // Ignore hidden files (like .DS_Store) or any file starting with a dot
+            if (file.startsWith(".")) return false
+            const fullPath = path.join(folderPath, file)
+            return existsSync(fullPath) && statSync(fullPath).isFile()
+         })
+         
+      return files.length > 0 && files.every(file => {
+         //Sometimes extentions can be upper case with toLowerCase() method we can aviod errors
+         const fileExtensions = extname(file).toLowerCase()
+         return VALID_IMAGE_EXTENSIONS.includes(fileExtensions)
+      })
+   }
+   // Function to prompt the user until a valid folder is selected
+   function promptForValidFolder(): string {

```

Comment:
Again, not sure here, but some of this could move to a new function in util.ts... Or maybe a function isn't needed for the prompting and you can just `while (true)` here in main.ts...